diff --git a/dist/build/webpack/plugins/flight-manifest-plugin.js b/dist/build/webpack/plugins/flight-manifest-plugin.js
--- a/dist/build/webpack/plugins/flight-manifest-plugin.js
+++ b/dist/build/webpack/plugins/flight-manifest-plugin.js
@@ -353,7 +353,7 @@ class ClientReferenceManifestPlugin {
                                 if (((_connection_module = connection.module) == null ? void 0 : _connection_module.constructor.name) === 'ConcatenatedModule') {
                                     const concatenatedMod = connection.module;
                                     const concatenatedModId = compilation.chunkGraph.getModuleId(concatenatedMod);
-                                    if (concatenatedModId) {
+                                    if (concatenatedModId !== null) {
                                         recordModule(concatenatedModId, clientEntryMod);
                                     }
                                 }
diff --git a/dist/esm/build/webpack/plugins/flight-manifest-plugin.js b/dist/esm/build/webpack/plugins/flight-manifest-plugin.js
--- a/dist/esm/build/webpack/plugins/flight-manifest-plugin.js
+++ b/dist/esm/build/webpack/plugins/flight-manifest-plugin.js
@@ -304,7 +304,7 @@ export class ClientReferenceManifestPlugin {
                                 if (((_connection_module = connection.module) == null ? void 0 : _connection_module.constructor.name) === 'ConcatenatedModule') {
                                     const concatenatedMod = connection.module;
                                     const concatenatedModId = compilation.chunkGraph.getModuleId(concatenatedMod);
-                                    if (concatenatedModId) {
+                                    if (concatenatedModId !== null) {
                                         recordModule(concatenatedModId, clientEntryMod);
                                     }
                                 }
